Direct comp effort

joaobrittoneto opened the pull request.

joaobrittoneto committed:
remove attempt of removing last error present

joaobrittoneto committed:
Library exclusive for dealing with matrix6 and msgs.proto

joaobrittoneto committed:
define node and publishers

joaobrittoneto committed:
rename proto file to uw_msgs

joaobrittoneto committed:
Publish CoG pose only, define Matrix6::Identity

Keep Mrb*(Mrb + Ma)^-1 in global variable. To be published

joaobrittoneto committed:
Publish CoG and compInertia in updateBegin

Make sure the connection with subscriber
(e.g. Thruster plugin) is established.
Avoid spamming every. Subscriber is not mandatory.

Needs a better event to do it just one time,
after connection with subscribe being created.

joaobrittoneto committed:
Matrix6 <-> gazebo_undewater::msgs::Matrix6

Requires Matrix3<->gazebo_underwater::msgs::Matrix3

joaobrittoneto committed:
Update Matrix6 constructors

joaobrittoneto commented:
@doudou, Do you know if there is an event that can be called by demand?
I could make it work by publishing the compensated matrix and CoG every UpdateBegin (what is not ideal since the value are constants)

doudou commented:
Given that the gazebo guys took a lot of things from ROS, you could have a look at the ability to get the last value written when subscribing to a topic.

joaobrittoneto committed:
Update uw_msgs. Uses only one message

joaobrittoneto committed:
Rename compensatedInertia to compInertiaEye

joaobrittoneto committed:
Apply M(M+Ma)^-1 direct to efforts

Do not consider in efforts of previous step.
Apply (M(M+Ma)^-a - I) to coriolis and gravity effect

joaobrittoneto committed:
Define operator == and != for Matrix6 and Vector6
